ocrd_network/ocrd_network/__init__.py

```diff
@@ -0,0 +1,17 @@
+# This network package is supposed to contain all the packages and modules to realize the network architecture:
+# https://user-images.githubusercontent.com/7795705/203554094-62ce135a-b367-49ba-9960-ffe1b7d39b2c.jpg
+
+# For reference, currently:
+# 1. The WebAPI is available here:
+# https://github.com/OCR-D/ocrd-webapi-implementation
+# 2. The RabbitMQ Library (i.e., utils) is available here:
+# https://github.com/OCR-D/ocrd-webapi-implementation/tree/main/ocrd_webapi/rabbitmq
+# 3. Some potentially more useful code to be adopted for the Processing Server/Worker is available here:
+# https://github.com/OCR-D/core/pull/884
+# 4. The Mets Server discussion/implementation is available here:
+# https://github.com/OCR-D/core/pull/966
+
+# Note: The Mets Server is still not placed on the architecture diagram and probably won't be a part of
+# the network package. The reason, Mets Server is tightly coupled with the `OcrdWorkspace`.
```

Comment:
Let's wait for input on this from @tdoan2010 as well.

Reply:
Robert's idea is the same as mine, so I completely agree.

A METS Server is basically the same as a Processing Worker. It listens to a specific queue (e.g., `mets-operations` queue), receives messages from the queue and updates the METS file accordingly.

Reply:
Yes, that's a good way to put it. A workspace-dedicated, specialised worker. But its lifetime is different than a Processing Worker – it gets set up and torn down for a /processing or /workflow request, not for the whole server uptime.

Reply:
@tdoan2010 Shall we track this somewhere else, so the discussion does not get lost after merge?

Reply:
> It listens to a specific queue (e.g., mets-operations queue), receives messages from the queue and updates the METS file accordingly.

1) I think this should be thought through properly. A separate message queue will be needed for each workspace instance. Is that ideal?
2) This would also mean that the METS Server has to use RabbitMQ Publisher internally. I don't think this would be ideal since parts of `ocrd_network` will be introduced inside `ocrd`.

Reply:
Of course, unless, the METS server is implemented in `ocrd_network` as an extension of the `ocrd.Workspace` instead of complicating the `ocrd.Workspace` as it currently feels in #966. But the main reason for #966 was also to consider the HPC cases where a RabbitMQ is no longer good.

Reply:
> 1. A separate message queue will be needed for each workspace instance. Is that ideal?
> 2. This would also mean that the METS Server has to use RabbitMQ Publisher internally

Definitely not ideal. Plus the fact that we would need to manage these queues (set up + tear down) with the workflow lifetime.

I'd rather have this separate and independent, as started in #966. Still, we should anticipate the changes required here (in the Processing Server, after this is merged and METS Server gets finished): The METS Server should be another parameter of the job (next to the workspace id/path), and (unless the Workflow Server can do this by itself) there would have to be an endpoint for starting them up and tearing it down.

Reply:
@bertsky 

> Definitely not ideal. Plus the fact that we would need to manage these queues (set up + tear down) with the workflow lifetime.

Exactly. Simply pushing/pulling everything from a queue is not good. 

> I'd rather have this separate and independent, as started in https://github.com/OCR-D/core/pull/966.

Agree.

> Still, we should anticipate the changes required here (in the Processing Server, after this is merged and METS Server gets finished)

Sure. The `ocrd_network` will continue getting developed. The reference WebAPI implementation will be transfered to `ocrd_network` ASAP once this PR is merged. 

> The METS Server should be another parameter of the job (next to the workspace id/path), and (unless the Workflow Server can do this by itself) there would have to be an endpoint for starting them up and tearing it down.

I think the Deployer agent should be separated by the Processing Server in the [OCR-D System Architecture](https://github.com/OCR-D/spec/blob/master/web_api.md#suggested-ocr-d-system-architecture). On an implementation level, it is already separated. Makes the code easier to follow. The Workflow Server can then start/stop the METS Servers for each workspace through the Deployer.

For the HPC environment, the batch script used to trigger the Nextflow workflow can be used to start the METS server before triggering the workflow, and stop it after the workflow finishes.

Reply:
> I think the Deployer agent should be separated by the Processing Server in the [OCR-D System Architecture](https://github.com/OCR-D/spec/blob/master/web_api.md#suggested-ocr-d-system-architecture). On an implementation level, it is already separated. Makes the code easier to follow. The Workflow Server can then start/stop the METS Servers for each workspace through the Deployer.

Yes, makes sense!

Reply:
continued in https://github.com/OCR-D/core/issues/1035